synthesizer/program/src/resources/credits.aleo

```diff
@@ -106,573 +149,515 @@ record credits:
-    // If the validator does not exist, this finalize scope will fail.
-    get committee[r0] into r2;
-    // Enforce that the validator is closed to new stakers.
-    assert.eq r2.is_open false;
-
-    // Check if the delegator is a validator.
-    contains committee[r1] into r3;
-    // Enforce the delegator is *not* a validator.
-    assert.eq r3 false;
-
-    /* End Committee */
-
-    /* Start Bonded */
-
-    // Get the bond state for the delegator, or fail if it does not exist.
-    get bonded[r1] into r4;
-    // Enforce that the delegator is bonded to the validator.
-    assert.eq r4.validator r0;
-
-    /* End Bonded */
-
-    /* Start Committee */
-
-    // Decrement the stake for the specified validator.
-    sub r2.microcredits r4.microcredits into r5;
-    // Construct the updated committee state.
-    cast r5 r2.is_open into r6 as committee_state;
-
-    // Get the number of delegators.
-    get metadata[aleo1qgqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqanmpl0] into r7;
-    // Decrement the number of delegators by one.
-    sub r7 1u32 into r8;
-
-    /* End Committee */
-
-    /* Start Unbond */
-
-    // Construct the initial unbond state.
-    cast 0u64 0u32 into r9 as unbond_state;
-    // Get the unbond state for the delegator, or default to the initial unbond state.
-    get.or_use unbonding[r1] r9 into r10;
-
-    // Increment the microcredits in the unbond state.
-    add r10.microcredits r4.microcredits into r11;
-    // Compute the height at which the unbonding will be complete, starting from the current block.
-    // Note: Calling unbond across multiple blocks before the unbonding is complete will reset the height each time.
-    add block.height 360u32 into r12;
-
-    // Construct the updated unbond state.
-    cast r11 r12 into r13 as unbond_state;
-
-    /* End Unbond */
-
-    /* Start Writes */
-
-    // Update the committee state for the specified validator.
-    set r6 into committee[r0];
-    // Remove the bond state for the delegator.
-    remove bonded[r1];
-    // Update the unbond state for the delegator.
-    set r13 into unbonding[r1];
-    // Update the number of delegators.
-    set r8 into metadata[aleo1qgqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqanmpl0];
-
-    /* End Writes */
-
-/**********************************************************************************************************************/
-
-// The `claim_unbond_public` function allows any staker to claim their microcredits
+// The `claim_unbond_public` function allows any address to run a crank to enable a staker to claim their microcredits
 // to their withdrawal address after the unbonding period.
 //
 // This function also removes the staker's withdrawal address if the staker no longer has any bonded balance.
 function claim_unbond_public:
+    // The address of the staker
+    input r0 as address.public;
+    // The number of microcredits to claim
+    input r1 as u64.public;
```

Comment:
What is the point of adding "The number of microcredits to claim"? You already enforce in finalize that this value must be equivalent to the entire unbond amount. Why not just implicitly unbond it all like the previous logic?

Reply:
@kyle-u410 My understanding is that this is a coinbase requirement for know exactly how many credits were claimed at a later date.

Reply:
It is important to know what the final claim amount is, but that should be present onchain when `claim_unbond_public` is finalized.

Computing the value to specify here would be annoying.

Additionally, even if the amount is specified here, there would still need to be sufficient block indexing to read the finalized result to know if the claim was successfull.

So I suspect the only practical result of requiring microcredits here would be an increased number of failed `claim_unbond_public` transactions.

Reply:
I'll remove this argument.